packages/sanity/src/core/releases/tool/detail/useBundleDocuments.ts

```diff
@@ -96,22 +92,14 @@ const getActiveReleaseDocumentsObservable = ({
           })
           .pipe(
             filter(Boolean),
-            switchMap((doc) =>
-              observableClient
-                .fetch(
-                  `*[_id in path("${getPublishedId(doc._id)}")]{_id}`,
-                  {},
-                  {tag: 'release-documents.check-existing'},
-                )
-                .pipe(
-                  switchMap((publishedDocumentExists) =>
-                    of({
-                      ...doc,
-                      publishedDocumentExists: !!publishedDocumentExists.length,
-                    }),
-                  ),
-                ),
-            ),
+            switchMap((doc) => {
+              return documentPreviewStore.unstable_observeDocumentPairAvailability(id).pipe(
+                map((availability) => ({
+                  ...doc,
+                  publishedDocumentExists: availability.published.available,
```

Comment:
(note for later) we should keep the document pristine and move away from writing `publishedDocumentExists` into the document, and rather return `{document: doc, publishedDocumentExists: availability.published.available}`